removal of 'udisks_state_add_loop' breaks gnome shell auto mount function

Hi,

originlly there was a call to ```udisks_state_add_loop``` routine in udisks manager (see [here](https://github.com/storaged-project/udisks/blob/udisks2/src/udiskslinuxmanager.c#L427))

Now it is missing and gnome shell auto mount stops working because ```should_automount()``` (see [here](https://git.gnome.org/browse/gnome-shell/tree/js/ui/components/automountManager.js?h=gnome-3-24#n161)) is now returning ```false``` as a due of a negative check result [here](https://git.gnome.org/browse/gvfs/tree/monitor/udisks2/gvfsudisks2volume.c?h=gnome-3-24#n468)

So please, any plans to restore the feature? 

*BTW*, there were nothing similar to ```udisks_state_remove/drop_loop```, is it by design?

Thanks

@vojtechtrefny 

Sorry, I cannot confirm right now, just cannot get sources compiled.

Tried to apply you patch over master branch and got
```
 udisksdaemon.c:292:29: error: format ‘%s’ expects argument of type ‘char *’, but argument 5 has type ‘int’ 
[-Werror=format=]
             udisks_warning ("Failed to load the '%s' libblockdev plugin",
```
For this simple case I replaced format specifier as it does not matter for this test, then get a couple of extra errors
```
udisksdaemon.c: In function ‘udisks_daemon_constructed’:
udisksdaemon.c:269:9: error: implicit declaration of function ‘bd_switch_init_checks’ [-Werror=implicit-function-declaration]
   ret = bd_switch_init_checks (FALSE, &error);
         ^~~~~~~~~~~~~~~~~~~~~
udisksdaemon.c:269:9: warning: nested extern declaration of ‘bd_switch_init_checks’ [-Wnested-externs]
In file included from udisksdaemon.c:26:0:
udisksdaemon.c:293:29: error: implicit declaration of function ‘bd_get_plugin_name’; did you mean ‘bd_get_plugin_soname’? [-Werror=implicit-function-declaration]
                             bd_get_plugin_name ((*plugin_p)->name));
                             ^
```

These do look as trivial (at least for me), thus I tried to checkout right from your commit number ```7ff5b7f6e5c25e622ffdce3eeee5fdb6607cb48e```...
Well, first error has gone, but recent two are still here.

Could you please advise or fix the code tree to get it compiled?

BTW. At the [old good] SVN times there was a rule of thumb, that any code in repository must at least compile, at any point of the time. Nowadays it does not apply anymore? Or am I missing something when compiling?

--- a/src/udiskslinuxmanager.c
+++ b/src/udiskslinuxmanager.c
@@ -401,6 +401,15 @@ handle_loop_setup (UDisksManager          *object,
 
   loop_device = g_strdup_printf ("/dev/%s", loop_name);
 
+  /* Update the udisks loop state file (/run/udisks2/loop) with information
+   * about the new loop device created by us.
+   */
+  udisks_state_add_loop (udisks_daemon_get_state (manager->daemon),
+                        loop_device,
+                        path,
+                        fd_statbuf_valid ? fd_statbuf.st_dev : 0,
+                        caller_uid);
+
   /* Determine the resulting object */
   error = NULL;
   wait_data.loop_device = loop_device;
@@ -423,16 +432,6 @@ handle_loop_setup (UDisksManager          *object,
   udisks_notice ("Set up loop device %s (backed by %s)",
                  loop_device,
                  path);
-
-  /* Update the udisks loop state file (/run/udisks2/loop) with information
-     about the new loop device created by us. We need to manually trigger
-     uevent for the device after this to update Loop interface properties. */
-  udisks_state_add_loop (udisks_daemon_get_state (manager->daemon),
-                        loop_device,
-                        path,
-                        fd_statbuf_valid ? fd_statbuf.st_dev : 0,
-                        caller_uid);
-  udisks_linux_block_object_trigger_uevent (UDISKS_LINUX_BLOCK_OBJECT (loop_object));
 
   udisks_manager_complete_loop_setup (object,
                                       invocation,

